dll/win32/shell32/CDefView.cpp

```diff
@@ -1196,6 +1258,19 @@ LRESULT CDefView::OnCreate(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandl
 
     m_hAccel = LoadAcceleratorsW(shell32_hInstance, MAKEINTRESOURCEW(IDA_SHELLVIEW));
 
+    BOOL bPreviousParentSpecial = m_isParentFolderSpecial;
+
+    // A folder is special if it is the Desktop folder, a network folder, or a Control Panel folder
+    m_isParentFolderSpecial = _ILIsDesktop(m_pidlParent) || _ILIsNetHood(m_pidlParent) 
```

Comment:
If we debug this a bit, for example add this line before it:
`pdump_always(m_pidlParent);`
Then it becomes obvious what the problem is when navigating to the control panel:
```
-------- pidl=0018FB28
[0018FB28] size=0020 type=1f attr=0x00000000 name="{20d04fe0-3aea-1069-a2d8-08002b30309d}" ((null),(null))
[0018FB3C] size=0020 type=1f attr=0x00000000 name="{21ec2020-3aea-1069-a2dd-08002b30309d}" ((null),(null))
```
So this pidl is in the form `"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}"`.
To fix this check, something like this should work:
`_ILIsControlPanel(ILFindLastID(m_pidlParent))`

Did you test the NetHood folder?

Reply:
```_ILIsControlPanel(ILFindLastID(m_pidlParent))``` worked, thanks.